src/scripts/features/step_definitions/registrar_nft_steps.ts

```diff
@@ -0,0 +1,310 @@
+                logger.debug(`result: ${JSON.stringify(result)}`)
+                expect(result.Ok).to.equal(principal)
+            }
+        }
+    }
+});
+
+
+Given(/^registrar ext_transfer action$/, async function (table) {
+    let dataTable = table.hashes()
+    for (let targetData of dataTable) {
+        const id = await get_token_id_by_name(targetData.name)
+
+        const caller = targetData.caller
+        let localRegistrar
+        if (targetData.caller != 'none') {
+            const identityInfo = identities.getIdentity(caller)
+            localRegistrar = createRegistrar(identityInfo)
+        } else {
+            localRegistrar = registrar
+        }
+        if (id != undefined) {
+
+            let from: User
+            if (targetData.from_type == "principal") {
+                from = {
+                    principal: identities.getPrincipal(targetData.from)
+                } as User
+            } else {
+                from = {
+                    address: targetData.from
+                } as User
+            }
+            let to: User
+            if (targetData.to_type == "principal") {
+                to = {
+                    principal: identities.getPrincipal(targetData.to)
+                } as User
+            } else {
+                to = {
+                    address: targetData.to
+                } as User
+            }
+
+            const result = await localRegistrar
+                .ext_transfer(get_transfer_request(from, to, id))
+            global_transfer_result_list.push(result)
+        }
+    }
+
+});
+When(/^all registrar ext_transfer is ok$/, function () {
+    for (let result of global_transfer_result_list) {
+        assert_remote_result(result)
+    }
+});
+Given(/^registrar ext_approve action$/, async function (table) {
+    let dataTable = table.hashes()
+    for (let targetData of dataTable) {
+        const spender = targetData.spender
+
+        const owner = await get_name_bear(targetData.name)
+        const token = await get_token_id_by_name(targetData.name)
+        logger.debug(`owner: ${JSON.stringify(owner)}`)
+        if (owner != undefined) {
+            const identityInfo = identities.getIdentity(owner)
+            let registrar = createRegistrar(identityInfo)
+            let approve_request = {
+                token: token,
+                subaccount: [],
+                allowance: BigInt(1),
+                spender: identities.getPrincipal(spender)
+            } as ApproveRequest
+            const _ = await registrar.ext_approve(approve_request)
+
+        }
+    }
+});
+When(/^last registrar ext_transfer result is err,expected err is "([^"]*)"$/, function (err) {
+
+    let last_result = global_transfer_result_list[global_transfer_result_list.length - 1]
+    if ('Err' in last_result) {
+        logger.debug(`last_result: ${JSON.stringify(JSON.stringify(last_result.Err).toString())}`)
+        let exp = JSON.stringify(last_result.Err)
```

Comment:
is there any problem to handle json format, why here is such code to handle \ { }